### ISSUE TITLE
Autocomplete Menu for `{{Variables}}` in the Command Field

### ISSUE THREAD

**Issue by Taitava** [Taitava/obsidian-shellcommands#33]
### Discussed in https://github.com/Taitava/obsidian-shellcommands/discussions/32

<div type='discussions-op-text'>

<sup>Originally posted by **FelipeRearden** September 20, 2021</sup>
@Taitava , It’s just an idea for us to talk :)

Add a feature that open a context menu when the user is writing `{{` in the command field 
- This context menu will provide all `{{variables}}` available in Shell Commands for an easy selection;
- After choosing the variable from the list, `}}` will be placed in the Command Field and the user continues to type. 
</div>

To summarise:
- A list of `{{variables}}` that pops up when a user types `{{`.
- It suggests all variables, and for variables that have _static_ parameters (e.g. `{{file_path}}` -> `{{file_path:absolute}}` and `{{file_path:relative}}`; and `{{folder_path}}` -> `{{folder_path:absolute}}` and `{{folder_path:relative}}`), it suggests also the arguments.
- Support custom entries via a YAML file in the plugin's folder.
- The autocomplete feature is powered by [kraaden/autocomplete](https://github.com/kraaden/autocomplete).

**Comment by Taitava** [comment]
I think this is now in quite a good shape.

The autocomplete list does not update when a user clicks the caret to another position, but it's a small thing. I've asked about it here: https://github.com/kraaden/autocomplete/issues/82#issuecomment-986198211 . I can look into it later, if the library gets support for this.

I'll make a beta version where this can be tested before an actual release.

**Comment by Taitava** [comment]
Released.
